### PR TITLE
feat(dto): add EmployeeDto for data transfer

### PR DESCRIPTION
**Pull request by Larisa-sh**
feat(Employee): hashCode and equals were overrided according to best practices. New field was added - LocalDateTime createdAt.

fix: rest package was renamed to controller

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -44,6 +53,23 @@
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>1.8</source> <!-- depending on your project -->
```
Comment (DzianisYermalovich) [inline pom.xml]: What does it mean 1.8?

File: src/main/java/com/example/springbootapp/dto/EmployeeDTO.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package com.example.springbootapp.dto;
+
+import com.example.springbootapp.entity.Gender;
+import com.example.springbootapp.entity.JobTitle;
+
+import java.util.Date;
+
+public class EmployeeDTO {
```
Comment (DzianisYermalovich) [inline src/main/java/com/example/springbootapp/dto/EmployeeDTO.java]: We agreed to use Java record here.

Comment (DzianisYermalovich) [inline src/main/java/com/example/springbootapp/dto/EmployeeDTO.java]: There are no equals and hashcode in this DTO.

File: src/main/java/com/example/springbootapp/mapper/EmployeeMapper.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import com.example.springbootapp.dto.EmployeeDTO;
+import com.example.springbootapp.entity.Employee;
+import org.mapstruct.Mapper;
+
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface EmployeeMapper {
+
+    EmployeeMapper INSTANCE = Mappers.getMapper(EmployeeMapper.class);
```
Comment (DzianisYermalovich) [inline src/main/java/com/example/springbootapp/mapper/EmployeeMapper.java]: Let's discuss it

File: src/main/java/com/example/springbootapp/rest/EmployeeController.java [hunk 4]
```diff
@@ -1,40 +1,50 @@
...
     @Autowired
     public EmployeeController(EmployeeService employeeService) {
         this.employeeService = employeeService;
     }
 
     @GetMapping
-    public List<Employee> getAllEmployees(){
-        return employeeService.showAllEmployees();
+    public List<EmployeeDTO> getAllEmployees(){
+        return employeeService.showAllEmployees().stream().map(employeeMapper::employeeToEmployeeDTO).collect(Collectors.toList());
```
Comment (DzianisYermalovich) [inline src/main/java/com/example/springbootapp/rest/EmployeeController.java]: You should avoid logic in controllers

File: src/main/java/com/example/springbootapp/mapper/EmployeeMapper.java (hunk 3 above)
Comment (Larisa-sh) [reply src/main/java/com/example/springbootapp/mapper/EmployeeMapper.java]: how to do it throught bean

**Review by DzianisYermalovich — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (Larisa-sh) [reply pom.xml]: it fixed

**Pull request merged**